src/array_partition.jl

```diff
@@ -503,3 +503,5 @@ function Base.convert(::Type{ArrayPartition{T,S}}, A::ArrayPartition{<:Any,<:NTu
         ntuple((@inline i -> convert(S.parameters[i], A.x[i])), Val(N))
     )
 end
+            
+Base.length(::Type{<:ArrayPartition{F,T}}) where {F,T <: Tuple} = T.parameters .|> length |> sum
```

Comment:
```suggestion
function Base.length(::Type{<:ArrayPartition{F,T}}) where {F,N,T <: NTuple{N,StaticArray}}
    return sum(map(length, T.parameters))
end
```
I think it's appropriate to give it a more narrow method signature.

Reply:
Does it need to be marked pure to act at compile time?

Reply:
okay thanks, on the parameter `N` where does that go in the actual type?  Is that `<:ArrayPartition{F,N,T}`?

Reply:
Ah, yes, Julia 1.7 can't constant-fold that, sadly. Adding `@generated` works though.

`N` is used to parametrize `NTuple` in upper bound for `T`.